nion/swift/Application.py

```diff
@@ -379,6 +379,8 @@ def show_open_project_dialog(self) -> None:
             filter_str = "Projects (*.nsproj);;Legacy Libraries (*.nslib);;All Files (*.*)"
             import_dir = ui.get_persistent_string("open_directory", ui.get_document_location())
             paths, selected_filter, selected_directory = ui.get_file_paths_dialog(_("Add Existing Library"), import_dir, filter_str)
+            if len(paths) == 0:
+                selected_directory = import_dir
             ui.set_persistent_string("open_directory", selected_directory)
```

Comment:
This seems like a bit of pretzel logic ... why not just move the `ui.set_persistent_string` inside the `if` statement on the next line?

Reply:
I could have something like:
`ui.set_persistent_string("open_directory", selected_directory if len(paths) else import_dir)` instead.
The main catch with the current logic setup is that the application saves the last folder the user used, and if we encapsulate `ui.set_persistent_string("open_directory", selected_directory)` in the if, the new directory will never be set. At which point it'd probably be simpler to use ui.get_document_location() as the import folder dialog does.

Reply:
I've made the minimal changes I suggested here:
https://github.com/nion-software/nionswift/pull/871

Do you see any problem with that approach?

Reply:
No that looks fine. I assume that was your original intent too, so I must have misunderstood your initial comment here.